File: persistence/src/main/java/io/sparkled/persistence/song/impl/query/DeleteRenderedStagePropsQuery.java
package io.sparkled.persistence.song.impl.query;

import io.sparkled.model.entity.RenderedStageProp;
import io.sparkled.persistence.PersistenceQuery;

import javax.persistence.EntityManager;
import javax.persistence.Query;
import java.util.logging.Logger;

public class DeleteRenderedStagePropsQuery implements PersistenceQuery<Void> {

    private static final Logger logger = Logger.getLogger(DeleteRenderedStagePropsQuery.class.getName());

    private final int songId;

    public DeleteRenderedStagePropsQuery(int songId) {
        this.songId = songId;
    }

    @Override
    public Void perform(EntityManager entityManager) {
        String className = RenderedStageProp.class.getSimpleName();
        Query query = entityManager.createQuery("delete from " + className + " where songId = :id");
        query.setParameter("id", songId);

        int deleted = query.executeUpdate();
        logger.info("Deleted " + deleted + " " + className + " record(s) for song " + songId);
        return null;
    }
}

File: persistence/src/main/kotlin/io/sparkled/persistence/stage/impl/query/DeleteRenderedStagePropsQuery.kt
package io.sparkled.persistence.stage.impl.query

import io.sparkled.model.entity.QRenderedStageProp.Companion.renderedStageProp
import io.sparkled.model.util.IdUtils.NO_IDS
import io.sparkled.persistence.PersistenceQuery
import io.sparkled.persistence.QueryFactory
import org.slf4j.LoggerFactory

class DeleteRenderedStagePropsQuery(renderedStagePropIds: Collection<Int>) : PersistenceQuery<Unit> {

    private val renderedStagePropIds: Collection<Int> =
        if (renderedStagePropIds.isEmpty()) NO_IDS else renderedStagePropIds

    override fun perform(queryFactory: QueryFactory) {
        val deleted = queryFactory
            .delete(renderedStageProp)
            .where(renderedStageProp.id.`in`(renderedStagePropIds))
            .execute()

        logger.info("Deleted {} rendered stage prop(s).", deleted)
    }

    companion object {
        private val logger = LoggerFactory.getLogger(DeleteRenderedStagePropsQuery::class.java)
    }
}
